test-framework/common/src/main/java/io/quarkus/test/ActivateSessionContext.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.test;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.concurrent.CompletionStage;
+
+import jakarta.interceptor.InterceptorBinding;
+
+/**
+ * Activates the session context before the intercepted method is called, and terminates the context when the method invocation
+ * completes (regardless of any exceptions being thrown).
+ * <p>
+ * If the context is already active, it's a noop - the context is neither activated nor deactivated.
+ * <p>
+ * Keep in mind that if the method returns an asynchronous type (such as {@link CompletionStage} then the session context is
+ * still terminate when the invocation completes and not at the time the asynchronous type is completed. Also note that session
+ * context is not propagated by MicroProfile Context Propagation.
+ * <p>
+ * This interceptor binding is only available in tests.
```

Comment:
Well, "intended for test" is not the same as "not available outside tests". I mean, for now I'd like to keep it for tests only because it doesn't make a lot of sense elsewhere.

Reply:
With this PR we already expose `Arc.container().sessionContext()` so the 'misuse' is very much possible already.
But let's keep it as-is for now :)

Reply:
Ofc, it is possible but we don't want to make it even easier...